lib/EventListener/UserEditListener.php

```diff
@@ -32,31 +39,136 @@ class UserEditListener implements EventSubscriberInterface
     /** @var UserController */
     protected $userController;
 
+    /** @var \eZ\Publish\Core\Repository\Repository */
+    private $repository;
+
+    /** @var \Symfony\Component\Routing\Router */
+    private $router;
+
+    /** @var \EzSystems\EzPlatformAdminUi\Form\Factory\FormFactory */
+    private $formFactory;
+
     /**
-     * @param ContentTypeService $contentTypeService
-     * @param ContentService $contentService
-     * @param UserController $userController
+     * @param \eZ\Publish\API\Repository\ContentTypeService $contentTypeService
+     * @param \eZ\Publish\API\Repository\ContentService $contentService
+     * @param \EzSystems\RepositoryFormsBundle\Controller\UserController $userController
+     * @param \eZ\Publish\Core\Repository\Repository $repository
+     * @param \Symfony\Component\Routing\Router $router
+     * @param \EzSystems\EzPlatformAdminUi\Form\Factory\FormFactory $formFactory
      */
     public function __construct(
         ContentTypeService $contentTypeService,
         ContentService $contentService,
-        UserController $userController
+        UserController $userController,
+        Repository $repository,
+        Router $router,
+        FormFactory $formFactory
     ) {
         $this->contentTypeService = $contentTypeService;
         $this->contentService = $contentService;
         $this->userController = $userController;
+        $this->repository = $repository;
+        $this->router = $router;
+        $this->formFactory = $formFactory;
     }
 
     public static function getSubscribedEvents()
     {
         return [
+            KernelEvents::REQUEST => [
+                ['beforeUserControllerCreate', 20],
+                ['beforeUserControllerEdit', 30],
+            ],
             KernelEvents::CONTROLLER => [
                 ['onControllerUserCreate', 20],
                 ['onControllerUserEdit', 30],
             ],
         ];
     }
 
+    /**
+     * @param \Symfony\Component\HttpKernel\Event\GetResponseEvent $event
+     *
+     * @throws \Exception
+     */
+    public function beforeUserControllerCreate(GetResponseEvent $event)
+    {
+        $request = $event->getRequest();
+        if (self::CONTENT_BASE_CREATE_ROUTE === $request->attributes->get('_route')) {
+            $form = $this->formFactory->contentEdit(null, 'content_create');
+            $this->repository->sudo(function () use ($form, $request) {
+                $form->handleRequest($request);
+            });
```

Comment:
:scream: more I read the code, more it scares me. This is not good approach. We use sudo for edge cases only. This PR shouldn't belong to them. Moreover handling entire request is quite a security hole.

Reply:
Yup, I discussed this with guys in our office, sudo is needed as routing is resolved before authorization.

>Moreover handling entire request is quite a security hole.

how is that different from all other places when we do this? (means, handling whole request).

Reply:
where are we handling whole request in sudo? sudo is mainly meant for specific api calls.

Reply:
I can always extract `id` and all other parameters from `$request` but this seems as dirty as this.